src/main/java/com/google/cloud/tools/app/Action.java

```diff
@@ -0,0 +1,137 @@
+
+  public Action(Map<Option, String> flags) {
+    this.flags = flags;
+  }
+
+  /**
+   * Executes the logic implemented by this {@link Action}.
+   *
+   * <p>For most actions, this is simply calling gcloud for the command that was supplied in the
+   * {@link Action} constructor.
+   *
+   * @return {@code true} if execution was successful, {@code false} if it was unsuccessful but no
+   * exception was thrown.
+   * @throws GCloudExecutionException if any error invoking or running gcloud occurred.
+   */
+  public boolean execute() throws GCloudExecutionException, IOException {
+    return processCaller.call();
+  }
+
+  @VisibleForTesting
+  public ProcessCaller getProcessCaller() {
+    return processCaller;
+  }
+
+  /**
+   * Checks if {@code flags} contains flags that are not accepted by the {@link Action} and if
+   * any flag's value is invalid.
+   *
+   * @throws InvalidFlagException if client-provided flags contain flags not accepted by the
+   * {@link Action}.
+   */
+  protected void checkFlags(Map<Option, String> clientProvidedFlags, Set<Option> acceptedFlags) {
+    for (Entry<Option, String> clientFlag : clientProvidedFlags.entrySet()) {
+      // Check for unrecognised flags.
+      if (!acceptedFlags.contains(clientFlag.getKey())) {
+        throw new InvalidFlagException(String.format(
+            "The %s flag is not recognised by the %s command.",
+            clientFlag.getKey().getLongForm(), this.getClass().getName()));
+      }
+
+      // Check for invalid values.
+      String flagValue = clientFlag.getValue();
+      if (clientFlag.getKey().getType().equals(Type.BOOLEAN)) {
+        // If boolean, either there is no value, or the value has to be "true" or "false".
+        if (!Strings.isNullOrEmpty(flagValue)) {
+          String flagValueLowerCase = flagValue.toLowerCase(Locale.ENGLISH);
+          if (!flagValueLowerCase.equals("true".toLowerCase(Locale.ENGLISH))
+              && !flagValueLowerCase.equals("false".toLowerCase(Locale.ENGLISH))) {
+            throw new InvalidFlagException(String.format(
+                "The value %s for flag %s is invalid.",
+                flagValue, clientFlag.getKey().getLongForm()));
+          }
+        }
+      } else if (clientFlag.getKey().getType().equals(Type.INTEGER)) {
+        // If integer, flag must have a value and be int parsable.
+        if (!Strings.isNullOrEmpty(flagValue)) {
+          try {
+            Integer.parseInt(flagValue);
+          } catch (NumberFormatException nfe) {
+            throw new InvalidFlagException(String.format(
+                "The value %s for flag %s is invalid.",
+                flagValue, clientFlag.getKey().getLongForm()));
+          }
+        } else {
+          throw new InvalidFlagException(
+              "Flag " + clientFlag.getKey().getLongForm() + " must have a value.");
+        }
+      }
+    }
+  }
+
+  /**
+   * Sets a new Cloud SDK location, in case it isn't installed in the default user's home
```

Comment:
I wouldn't bother specifying the case here. This is just a way to manually configure the Cloud SDK path. The use cases can be varied.